webview/src/experiments/components/table/Row.tsx

```diff
@@ -92,6 +96,46 @@ const getMultiSelectMenuOptions = (selectedRowsList: RowProp[]) => {
       MessageFromWebviewType.REMOVE_EXPERIMENT,
       hideRemoveOption,
       true
+    ),
+    {
+      divider: true,
+      id: 'clear-selection',
+      keyboardShortcut: 'Esc',
+      label: 'Clear row selection'
+    }
+  ]
+}
+
+const getWorkspaceOptions = (
+  withId: (
+    label: string,
+    type: MessageFromWebviewType,
+    hidden?: boolean,
+    divider?: boolean
+  ) => MessagesMenuOptionProps,
+  isWorkspace: boolean,
+  projectHasCheckpoints: boolean,
+  hideVaryAndRun: boolean,
+  depth: number
+) => {
+  const isNotCheckpoint = depth <= 1 || isWorkspace
+
+  return [
+    withId(
+      projectHasCheckpoints ? 'Modify and Resume' : 'Modify and Run',
+      MessageFromWebviewType.VARY_EXPERIMENT_PARAMS_AND_RUN,
+      !isNotCheckpoint,
```

Comment:
[C] Might be easier to read if you change the condition to `isCheckpoint` instead of the double negative.

Reply:
try and simplify this boolean logic wherever you can, it is getting pretty difficult to understand at first glance.

Reply:
I don't like it either, but I have a feeling this refactoring will take some time. I can tackle it in the next PR, if that's ok.